config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -204,8 +205,15 @@ private static String hasIpAddress(String ipAddressExpression) {
 	public final class ExpressionInterceptUrlRegistry extends
 			ExpressionUrlAuthorizationConfigurer<H>.AbstractInterceptUrlRegistry<ExpressionInterceptUrlRegistry, AuthorizedUrl> {
 
+		private String rolePrefix = "ROLE_";
+
 		private ExpressionInterceptUrlRegistry(ApplicationContext context) {
 			setApplicationContext(context);
+			try {
+				this.rolePrefix = context.getBean(GrantedAuthorityDefaults.class).getRolePrefix();
```

Comment:
Instead of catching an exception, which can be costly, please call `ApplicationContext#getBeanNamesForType`. This will return an empty array if there is no `GrantedAuthorityDefaults` bean.